src/inject/dynamic-theme/index.ts

```diff
@@ -54,6 +57,19 @@ function createOrUpdateScript(className: string, root: ParentNode = document.hea
     return element;
 }
 
+/**
+ * Note: This function is used only with MV3.
+ * String passed as src parameter must be included in web_accessible_resources manifest key.
+ */
+function createScriptMV3(src: string, args: any = undefined) {
+    const element = document.createElement('script');
+    element.src = chrome.runtime.getURL(src);
```

Comment:
So, we had to revert earlier changes to this to be MV3-compliant because the loading of the script was too slow and "not blocking". Is that the same case here? Otherwise this isn't a valid way of injecting the proxy. Styles can change between the initial run of going trough styles and this script actually being loaded.

Reply:
> the loading of the script was too slow and "not blocking". Is that the same case here?

Unfortunately, I think so. I believe this is the limitation of current API in general. I am aware of multiple extensions which would benefit from injecting a script into the real/non-isolated page context early on in the page load process, but as far as I know, this feature does not exist right now. The last I checked, multiple widely used commercial and non-commercial extensions will cease to run in January 2023 once MV3 becomes mandatory. A larger number of extensions will regress in some way.

I see two potential solutions:
 - creating a separate CS solely for injecting this proxy. The race condition will still be there, but the proxy will have a much greater chance of winning.
 - I remember there was a discussion about adding an extra parameter to content scripts registrations in manifest. This proposed parameter would have two values "isolated world" (default) and "the real page world". This probably would be the perfect solution, but I can't find the relevant discussion.

Reply:
Converted to draft until I we resolve this.

Reply:
> A larger number of extensions will regress in some way.

Yup a big unfortunate for the little benefit that MV3 will add. But hey we must somehow conform to it, so just let get over with it 😄.

So I think we can use a variation of your potential solution. 




> * creating a separate CS solely for injecting this proxy. The race condition will still be there, but the proxy will have a much greater chance of winning.

I think this is the "best solution" to mitigate the timing issue, but I think we should also add mechanism to fire a event for when we injected the style(and thus from there-on were watching for style modifications) and when we catch that event we check if we already did the initial rounds of style checking and then just blatantly again go trough all styles. It's really inefficient but the best thing to do in this case.

Reply:
I prototyped the proposed solution and had hit a blocker: `injectScript` requires an argument which is known only by content script. We could default to `true` and then just request this data from CS and "undo" the extra proxies. This seems like the most logical solution.

Also, we can attempt to inject proxy from both places (a dedicated CS and the current CS) and just see whichever wins.

Reply:
> (a dedicated CS 

We probably should re-use the fallback.js, Otherwise the timing between executing content scripts becomes too big. We don't want to delay the first paint too much.